lib/core/src/math/utils.rs

```diff
@@ -12,18 +17,52 @@ pub trait IntoCSSMatrix {
 impl<T : RealField> IntoCSSMatrix for Matrix4<T> {
     fn into_css_matrix(&self) -> String {
         let mut iter = self.iter();
-        let item = iter.next().expect("Matrix4 should have the first item");
-        let acc = format!("{}", item);
-        let ret = iter.fold(acc, |acc, item| format!("{}, {}", acc, item));
+        let item     = iter.next().expect("Matrix4 should have the first item");
+        let acc      = format!("{}", item);
+        let ret      = iter.fold(acc, |acc, item| format!("{}, {}", acc, item));
         format!("matrix3d({})", ret)
     }
 }
 
+
+// ========================
+// === Float32ArrayView ===
+// ========================
+
+/// A Float32Array view created from `IntoFloat32ArrayView`.
+#[derive(Shrinkwrap)]
+pub struct Float32ArrayView<'a> {
+    #[shrinkwrap(main_field)]
+    array   : Float32Array,
+    phantom : PhantomData<&'a Float32Array>
+}
+
+pub trait IntoFloat32ArrayView {
+    /// # Safety
+    /// Views into WebAssembly memory are only valid so long as the backing buffer isn't resized in JS. Once this function is called any future calls to Box::new (or malloc of any form) may cause the returned value here to be invalidated. Use with caution!
+    ///
+    /// Additionally the returned object can be safely mutated but the input slice isn't guaranteed to be mutable.
+    ///
+    /// Finally, the returned object is disconnected from the input slice's lifetime, so there's no guarantee that the data is read at the right time.
+    unsafe fn into_float32_array_view(&self) -> Float32ArrayView<'_>;
+}
+
+impl IntoFloat32ArrayView for Matrix4<f32> {
+    unsafe fn into_float32_array_view(&self) -> Float32ArrayView<'_> {
+        // Note [2D array to 1D array]
+        let matrix = self.as_ref();
+        let matrix = matrix as *const [[f32; 4]; 4];
+        let array = Float32Array::view(&*(matrix as *const [f32; 16]));
+        let phantom = PhantomData;
+        Float32ArrayView { array, phantom }
+    }
+}
```

Comment:
This is the only implementation of this trait and unsafe function - in such case, why do we want to keep it as a trait, not just as a single unsafe function? Maybe it's a good design, but Id like to confirm it.

Reply:
The reasoning here is that we might want to use this view on other objects as well and because I think it's prettier to use this function as a method-like fashion like `matrix.into_float32_array_view()`.